neo/Network/P2P/ProtocolHandler.cs

```diff
@@ -299,6 +332,28 @@ private void OnVersionMessageReceived(VersionPayload payload)
             Context.Parent.Tell(payload);
         }
 
+        private void OnTimer()
+        {
+            RefreshPendingKnownHashes();
+        }
+
+        protected override void PostStop()
+        {
+            timer.CancelIfNotNull();
+            base.PostStop();
+        }
+
+        private void RefreshPendingKnownHashes()
+        {
+            while (pendingKnownHashes.Count > 0)
```

Comment:
Will not this While be an eternal loop?

Isnt it just a loop to iterate over the pending and delete those who exceeded time?

Reply:
For me, `while` is clearer as the  `pendingKnownHashes` might be changed in the loop. It doesn't continue running if the first item is not exceed `PendingTimeout`.

Reply:
I understood now. Thanks,